src/lib.rs

```diff
@@ -224,8 +224,37 @@
 //!     mass: Vec<f32>
 //! }
 //! ```
-//! 
+//!
 //! All helper structs will be also nested, for example `PointSlice` will be nested in `ParticleSlice`.
+//!
+//! # Use in a generic context
+//!
+//! `StructOfArray` does not provide a set of common operations by default. Thus if you wanted to use a `StructOfArray`
+//! type in a generic context, there is no way to guarantee to the type system that any methods are available.
+//!
+//! If the `generic_traits` feature is enabled, the attribute macro `#[generate_traits]` will generate
+//! implementations of [`SoAVec`], [`SoASlice`], and [`SoASliceMut`] for the respective `Vec`, `Slice`
+//! and `SliceMut` types.
+//!
+//! These rely on GATs, and so require Rust 1.65 or newer, and so this feature is disabled by default.
+//! Even when enabled, trait implementations are only generated by opting in with the `#[generate_traits]`
+//! attribute.
+//!
+//! ```ignore
+//! # mod cheese {
+//! # use soa_derive::{StructOfArray, prelude::*};
+//! #[derive(StructOfArray)]
+//! #[generate_traits]
```

Comment:
not sure if this is up to date, but what would be the cost of always generating the trait implementations? 

And if  the cost is too high to always enable them, then this should be scoped with something like  `#[soa_derive(generate_traits)]`